upup/pkg/fi/cloudup/openstacktasks/instance.go

```diff
@@ -97,30 +100,55 @@ func (e *Instance) Find(c *fi.Context) (*Instance, error) {
 	if e == nil || e.Name == nil {
 		return nil, nil
 	}
+	tmp := strings.Replace(strings.ToLower(fi.StringValue(e.GroupName)), "_", "-", -1)
+	groupName := strings.Replace(tmp, ".", "-", -1)
 	serverPage, err := servers.List(c.Cloud.(openstack.OpenstackCloud).ComputeClient(), servers.ListOpts{
-		Name: fmt.Sprintf("^%s$", fi.StringValue(e.Name)),
+		Name: fmt.Sprintf("^%s", groupName),
 	}).AllPages()
 	if err != nil {
-		return nil, fmt.Errorf("error finding server with name %s: %v", fi.StringValue(e.Name), err)
+		return nil, fmt.Errorf("error listing servers: %v", err)
 	}
 	serverList, err := servers.ExtractServers(serverPage)
 	if err != nil {
 		return nil, fmt.Errorf("error extracting server page: %v", err)
 	}
-	if len(serverList) == 0 {
+
+	filteredList := []servers.Server{}
+	for _, server := range serverList {
+		val, ok := server.Metadata["k8s"]
+		if !ok || val != fi.StringValue(e.ClusterName) {
+			continue
+		}
+		metadataName := ""
+		val, ok = server.Metadata[openstack.TagKopsName]
+		if ok {
+			metadataName = val
+		}
+		_, detachTag := server.Metadata[openstack.TagNameDetach]
```

Comment:
I don't know how we can make this one generic, though :-)  We _could_ pass in `ignoreMatching map[string]string` but it's starting to get a little too abstract, perhaps :-)  If you do change the others, I think it's fine to leave Detach as a special-case :-)

Reply:
the problem in openstack is that servergroups does not support things that for instance AWS instancegroups do support. This thing is much more difficult than in AWS